### PR TITLE
feat: Add loading prop for Input.Search, close #14538

### LINKED ISSUE #14538: Input.Search loading
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
网络慢时，防止用户继续点击去请求

### What does the proposed API look like?
输入关键字进行模糊查询，然后进入loading状态不可点击

### PR DESCRIPTION
**Pull request by dengfuping**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch.
Pull request will be merged after one of collaborators approve.
Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

- Add `loading` prop for `Input.Search` to avoid repetitive search before response returns. 
- Close #14538.
  
### API Realization (Optional if not new feature)

> 1. Basic thought of solution and other optional proposal.

- Display loading status when loading is true for Input.Search. And search and press enter event will not be effective.

> 2. List final API realization and usage sample.

- API realization:

| Property | Description | Type | Default |
| -------- | ----------- | ---- | ------- |
| loading | Searching status added in 3.15.0 | boolean | false |

- Usage sample:

````jsx
import { Input } from 'antd';

const Search = Input.Search;

class SearchDemo extends React.Component {
  state = {
    loading1: false,
    loading2: false,
    loading3: false,
  };

  handleSearch = (key, value) => {
    console.log(value);
    this.setState({
      [key]: true,
    });
    setTimeout(() => {
      this.setState({
        [key]: false,
      });
    }, 3000);
  }

  render () {
    const { loading1, loading2, loading3 } = this.state;
    return (
      <div>
        <Search
          placeholder="input search text"
          onSearch={value => this.handleSearch('loading1', value)}
          style={{ width: 200 }}
          loading={loading1}
        />
        <br /><br />
        <Search
          placeholder="input search text"
          onSearch={value => this.handleSearch('loading2', value)}
          enterButton
          loading={loading2}
        />
        <br /><br />
        <Search
          placeholder="input search text"
          enterButton="Search"
          size="large"
          onSearch={value => this.handleSearch('loading3', value)}
          loading={loading3}
        />
      </div>
    );
  }
}

ReactDOM.render( <SearchDemo />, mountNode);
````

> 3. GIF or snapshot should be provided if includes UI/interactive modification.

image

### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?

- Yes. `loading` status for Input.Search maybe affect user when it is used.

> 2. What will say in changelog?

> 3. Does this PR contains potential break change or other risk?

- No, just add a new property.

### Changelog description (Optional if not new feature)

> 1. English description

- 🌟 Add `loading` prop to set searching status for Input.Search.

> 2. Chinese description (optional)

- 🌟 Input.Search 组件新增 `loading` 属性用于设置是否处于搜索中状态。

### Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### Additional Plan? (Optional if not new feature)

### CODE REVIEW

**Comment by zombieJ** [comment]
Any process?

File: components/input/demo/search-input.md [hunk 1]
```diff
@@ -7,38 +7,65 @@ title:
...
+          size="large"
+          onSearch={value => this.handleSearch('loading3', value)}
+          loading={loading3}
+        />
+      </div>
+    );
+  }
+}
+
+ReactDOM.render( <SearchDemo />, mountNode);
```
Comment (afc163) [inline components/input/demo/search-input.md]: loading 的例子单独列出来吧，不用动原来的例子。

File: components/input/demo/search-input.md [hunk 2]
```diff
@@ -7,38 +7,65 @@ title:
...
+          style={{ width: 200 }}
+          loading={loading1}
+        />
+        <br /><br />
+        <Search
+          placeholder="input search text"
+          onSearch={value => this.handleSearch('loading2', value)}
+          enterButton
+          loading={loading2}
+        />
```
Comment (afc163) [inline components/input/demo/search-input.md]: 这个例子按钮宽度不应该变动。

File: components/input/demo/search-input.md [hunk 3]
```diff
@@ -7,38 +7,65 @@ title:
...
+        <br /><br />
+        <Search
+          placeholder="input search text"
+          enterButton="Search"
+          size="large"
+          onSearch={value => this.handleSearch('loading3', value)}
+          loading={loading3}
+        />
```
Comment (afc163) [inline components/input/demo/search-input.md]: 切换 loading 时，按钮宽度变化应该 transition。

Comment (afc163) [reply components/input/demo/search-input.md]: loading 效果统一成去掉文字的 loading 图标好了。

**Pull request closed without merging**